### PR TITLE
JSDK-2687-signalingRegion Property on LocalParticipant

### PR DESCRIPTION
**Pull request by PikaJoyce**
Added signalingRegion property on the Room.localParticipant object. Implemented integration tests and fixed the LocalParticipantV2 mock for unit testing.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: lib/signaling/v2/localparticipant.js [hunk 1]
```diff
@@ -71,10 +75,26 @@ class LocalParticipantV2 extends LocalParticipantSignaling {
...
       }
     });
   }
 
+  /**
+   * Set the {@link signalingRegion}.
+   * @param {signalingRegion} signalingRegion.
+   */
+  setSignalingRegion(signalingRegion) {
+    if (!isDeepEqual(this._signalingRegion, signalingRegion)) {
```
Comment (manjeshbhargav) [inline lib/signaling/v2/localparticipant.js]:
```suggestion
    if (!this._signalingRegion) {
```

File: lib/signaling/v2/localparticipant.js [hunk 2]
```diff
@@ -71,10 +75,26 @@ class LocalParticipantV2 extends LocalParticipantSignaling {
...
+        enumerable: true,
+        get() {
+          return this._signalingRegion;
+        }
       }
     });
   }
 
+  /**
+   * Set the {@link signalingRegion}.
```
Comment (manjeshbhargav) [inline lib/signaling/v2/localparticipant.js]:
```suggestion
   * Set the signaling region.
```

File: lib/signaling/v2/localparticipant.js [hunk 3]
```diff
@@ -71,10 +75,26 @@ class LocalParticipantV2 extends LocalParticipantSignaling {
...
+   * @param {signalingRegion} signalingRegion.
```
Comment (manjeshbhargav) [inline lib/signaling/v2/localparticipant.js]:
```suggestion
   * @param {string} signalingRegion.
```

File: test/integration/spec/connect.js [hunk 4]
```diff
@@ -258,6 +258,63 @@ describe('connect', function() {
     });
   });
 
+  describe('default(gll) signaling region on LocalParticipant', () => {
```
Comment (manjeshbhargav) [inline test/integration/spec/connect.js]: @PikaJoyce I think its great that you were able to write tests to verify your implementation. However, I just realized there is already a test suite for signaling regions. Please search for `signaling region`, and then just add an `assert.equals` check for the signaling region at the end of the test function.

File: test/integration/spec/connect.js [hunk 5]
```diff
@@ -250,6 +250,8 @@ describe('connect', function() {
               assert(error instanceof SignalingConnectionError);
             } else {
               assert.equal(error, null);
+              assert.equal(typeof room.localParticipant.signalingRegion, 'string');
```
Comment (manjeshbhargav) [inline test/integration/spec/connect.js]:
```suggestion
              if (typeof region !== 'string' || region === 'gll') {
                assert.equal(typeof room.localParticipant.signalingRegion, 'string');
              } else {
                assert.equal(room.localParticipant.signalingRegion, region);
              }
```

**Comment by manjeshbhargav** [comment]
@PikaJoyce One more thing... please add a CHANGELOG.md entry describing your changes.

**Comment by PikaJoyce** [comment]
Reopening in #903

**Pull request closed without merging**